Adds note about public directory (#17203)
vercel/next.js: docs/basic-features/static-file-serving.md

> **Note**: Be sure to not have a static file with the same name as a file in the `pages/` directory, as this will result in an error.
>
> Read more: <http://err.sh/next.js/conflicting-public-file-page>

> > **Note**: Only assets that are in the `public` directory at [build time](/docs/api-reference/cli.md#build) will be served by Next.js. Files added to the directory after build time won't be discoverable until your code is rebuilt. We recommend using a third party service like [AWS S3](https://aws.amazon.com/s3/) for persistant file storage.

Reviewer: ```suggestion
> **Note**: Only assets that are in the `public` directory at [build time](/docs/api-reference/cli.md#build) will be served by Next.js. Files added to the directory after build time won't be discoverable until your code is rebuilt. We recommend using a third party service like [AWS S3](https://aws.amazon.com/s3/) for persistent file storage.
```

> > **Note**: Only assets that are in the `public` directory at [build time](/docs/api-reference/cli.md#build) will be served by Next.js. Files added to the directory after build time won't be discoverable until your code is rebuilt. We recommend using a third party service like [AWS S3](https://aws.amazon.com/s3/) for persistent file storage.

Reviewer 2: Adding files after build time won't make them available for a next build.

```suggestion
> **Note**: Only assets that are in the `public` directory at [build time](/docs/api-reference/cli.md#build) will be served by Next.js. Files added at runtime won't be available. We recommend using a third party service like [AWS S3](https://aws.amazon.com/s3/) for persistent file storage.
```

Author: Good point. Honestly I'm a little confused how people are continuing to update their app in these situations :shrug:
